tools/dataset_converters/yolo2coco.py

```diff
@@ -0,0 +1,221 @@
+            test_txt_path):
+        print(
+            'these files are located, need to organize the data accordingly.')
+        to_categorize = True
+    else:
+        print('these files are not located, no need to organize separately.')
+        to_categorize = False
+
+    with open(yolo_class_txt) as f:
+        classes = f.read().strip().split()
+
+    indices = os.listdir(yolo_image_dir)
+
+    if to_categorize:
+        train_dataset = {'images': [], 'annotations': [], 'categories': []}
+        val_dataset = {'images': [], 'annotations': [], 'categories': []}
+        test_dataset = {'images': [], 'annotations': [], 'categories': []}
+
+        # categories id starts from 0
+        for i, cls in enumerate(classes, 0):
+            train_dataset['categories'].append({'id': i, 'name': cls})
+            val_dataset['categories'].append({'id': i, 'name': cls})
+            test_dataset['categories'].append({'id': i, 'name': cls})
+        train_img, val_img, test_img = organize_by_existing_files(image_dir)
+    else:
+        dataset = {'images': [], 'annotations': [], 'categories': []}
+        for i, cls in enumerate(classes, 0):
+            dataset['categories'].append({'id': i, 'name': cls})
+
+    obj_count = 0
+    for idx, image in enumerate(mmengine.track_iter_progress(indices)):
+        # support both .jpg, .png, and .jpeg
+        img_suffix = osp.splitext(image)[1].lower()
+        if img_suffix not in ['.jpg', '.png', '.jpeg']:
+            raise Exception(
+                "Only supports '.jpg', '.png', and '.jpeg' image formats")
+        img_name = osp.splitext(image)[0]
+        img_info_dict, image_height, image_width = get_image_info(
+            yolo_image_dir, idx, image)
+
+        if to_categorize:
+            if image in train_img:
+                dataset = train_dataset
+            elif image in val_img:
+                dataset = val_dataset
+            elif image in test_img:
+                dataset = test_dataset
+
+        dataset['images'].append(img_info_dict)
+
+        label_path = osp.join(yolo_label_dir, img_name) + '.txt'
+        if not osp.exists(label_path):
+            # if current image is not annotated
+            print(f'WARNING: {label_path} does not exist. Skipped.')
+            continue
+
+        with open(label_path) as f:
+            labels = f.readlines()
+            for label in labels:
+                coco_info, obj_count = convert_bbox_info(
+                    label, idx, obj_count, image_height, image_width)
+                dataset['annotations'].append(coco_info)
+                obj_count += 1
+
+    # saving results to result json in res_folder
+    res_folder = osp.join(image_dir, 'coco_format')
+    if not osp.exists(res_folder):
+        os.makedirs(res_folder)
+
+    if to_categorize:
+        for category in ['train', 'val', 'test']:
+            out_file = osp.join(image_dir, f'coco_format/{category}.json')
```

Comment:
coco_format/{category}.json -> annotations/{category}.json

Reply:
> coco_format/{category}.json -> annotations/{category}.json

got it!